[LimZiJia] iP

LimZiJia opened the pull request:
____  _  _____ ____ ___ ____ ___ 
                                        / ___|| |/ /_ _| __ )_ _|  _ \_ _|
                                        \___ \| ' / | ||  _ \| || | | | | 
                                         ___) | . \ | || |_) | || |_| | | 
                                        |____/|_|\_\___|____/___|____/___|
# Skibidi Bot

> “[Cats] are blissfully unaware that they have only a finite time in which to finish their ‘to do’ list.” ― Jon Edgell

But Skibidi knows that there are deadlines and shit and Skibidi is here to help. It is:
* Text-based
* Easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,

1. Download it from [here](https://github.com/LimZiJia/ip/releases/tag/v0.1)
2. Run with ```java -jar duke-v0.1```*
3. Add your tasks.
4. Let Skibidi manage your tasks for you 🚽 🔥 🚽 🔥 🚽  

And it is **FREE!**

Features:
- [x] Managing Tasks
- [x] Managing Deadlines
- [x] Managing Events
- [ ] Reminders (coming soon)
___
If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```Java
public static void main(String[] args) {
        new Duke("/data/", "duke.txt").run();
}
```

\* Please make sure the version number is correct

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

LimZiJia committed:
Update .gitignore

Specifically asked not to commit any .class files

LimZiJia committed:
Update logo

Changing name of logo

LimZiJia committed:
Implemented greeting and bye statements.

New class called Skibidi to have all the functions and variables so that Duke only has the main runtime to worry about.

LimZiJia committed:
Added echo

Made the bot echo users' input before exiting when the user says 'bye'

LimZiJia committed:
List

Added the ability to store whatever the user was typing, and display when user types 'list'.

LimZiJia committed:
Implemented Markings

Made a task class to handle tasks. Made a function just for string comprehension. Bugs: have not accounted for integer passing exception.

LimZiJia committed:
Fixed invalid number parsing bug

Caught NumberFormatException to print "Not a valid number!"

LimZiJia committed:
Create subclasses

Made each subclass have fields that it needs and format string properly

LimZiJia committed:
Finish implementing tasks

Changed name of echo to chat, because it is more appropriate now. Added function addTask to slice and add tasks to the list.

LimZiJia committed:
Testing

Created the input file and expected to test that the program is working. Running the .bat somehow shows that the logo is different but I have no idea why.

LimZiJia committed:
Duke exceptions

Created a few exceptions that I think I will encounter with Duke

LimZiJia committed:
Handled task input exceptions

refactored some code and added exceptions handling

LimZiJia committed:
Implemented ArrayList

Changed they array of tasks to an array list of tasks. Removed the counter tasks.

LimZiJia committed:
Added delete

Added the option to delete from list. Rewrote some code to make use of exceptions to catch instead of another if statement for index error.

LimZiJia committed:
Refactored inputComprehension

Broke the function up into three parts so that it is easier to read.

LimZiJia committed:
Add save() method

The save method makes sure that the file and directory are in order.

That is the ./data/ directory exists and that the ./data/duke.txt does not.

We do not want the txt to exist because we want to write a new version everytime.

LimZiJia committed:
Add load()

Now the bot loads up a file if it exists.

Changed the formatting for the saved file to be comma separated.

Added a stringToTask parser.

Added more constructors and string format in the various tasks.

LimZiJia committed:
Change date time format

Changed from accepting any string to only parse using the ISO_LOCAL_DATE format.

Changed the saved format for date time to ISO_LOCAL_DATE.

Changed toString() of tasks to pattern "MMM dd yyyy"

LimZiJia committed:
Fix errors

Rebasing the branch made me lose some code in Task.java, Todo.java and Event.java. I have written them now.

Added stricter restrictions for parsing, because date input are now of fixed length.

LimZiJia committed:
Add exceptions

Adding more date time parsing related exceptions to fit the new parsing rules.

LimZiJia committed:
Merge branch 'branch-Level-7'

LimZiJia committed:
Merge branch 'branch-Level-8'

LimZiJia committed:
Fix bugs

Realised date from should be strictly before date to. Added new exception

LimZiJia committed:
Update test

After implementing save and new date time format, I updated the tests to suite my situation now.

LimZiJia committed:
Add classes

First try at it. Have not tested.

LimZiJia committed:
Test branch

Ran the tests that I have written before, fixed very minor hardcoding

LimZiJia committed:
Organise pakages

Grouped classes into some packages.

LimZiJia committed:
Change test

With packages, I tried to modify the directories the test script is accessing.

I failed to make it work.

LimZiJia committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

LimZiJia committed:
add Junit

Written tests that covers 80% of methods

LimZiJia committed:
Update build.gradle

Minor adjustments to shadowJar

LimZiJia committed:
Add javadoc

Added Javadoc to all public methods and classes with a explanations for private fields too.

LimZiJia committed:
Adhere to check style

Updated my code to fit with check style. Mainly that block of indent before commas instead of after.

LimZiJia committed:
Add find() function

Added the functionality to find tasks containing the substring.

Minor check style fixes.

LimZiJia committed:
Merge branch 'branch-A-CodingStandard'

LimZiJia committed:
Merge branch 'branch-Level-9'

LimZiJia committed:
Update JavaDoc

Added method description for find() in TaskList

LimZiJia committed:
Add checkstyle

Refactored some code and fixed checkstyle errors

Lin-Shuang-Shuang commented on src/main/com/duke/Storage.java:
```diff
@@ -0,0 +1,119 @@
+package com.duke;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.tasks.Deadline;
+import com.tasks.Event;
+import com.tasks.Task;
+import com.tasks.TaskList;
+import com.tasks.Todo;
```
I like that you separated your import statements as it looks neat and tidy.

Lin-Shuang-Shuang commented on src/test/duke/ParserTest.java:
```diff
@@ -0,0 +1,293 @@
+package duke;
```
I like how detailed your test cases are. They are very well-crafted.

Lin-Shuang-Shuang commented on src/main/com/duke/Storage.java:
```diff
@@ -0,0 +1,119 @@
+     * @param s Line from the saved file representing a Task.
+     * @return Task that is derived from the saved string representation.
+     */
+    private Task stringToTask(String s) {
```
Could this method name be renamed to something better? Perhaps make it a verb?

Lin-Shuang-Shuang commented on src/main/com/duke/Storage.java:
```diff
@@ -0,0 +1,119 @@
+        if (!Files.exists(pathDir)) {
+            Files.createDirectories(pathDir);
+        }
+        // Check if the save file exists
```
I like that you add comments to your code so readers can understand it better.

Lin-Shuang-Shuang commented on src/main/com/duke/Ui.java:
```diff
@@ -0,0 +1,63 @@
+    /**
+     * Says goodbye to the user.
+     */
+    public void bye() {
```
Perhaps you could change the method name to a verb like exit?

Lin-Shuang-Shuang reviewed (changes requested):
LGTM. Overall, your code is well written and easy to read. Some changes to method name requested.

LimZiJia committed:
Rename com to duke

Renamed directory names to be more appropriate.

Also added JavaFX into build.gradle

LimZiJia committed:
Draft JavaFX GUI

Did everything up to part 3 of the guide.

LimZiJia committed:
Add GUI

Tried to do FXML.

Main 'bugs' cannot expand hbox according to children. And cannot flip according to who is talking.

LimZiJia committed:
Remove bug

Removed an experimental line that I did not intend to commit.

LimZiJia committed:
Implement Java assertions

The are no assertions currently.

Without assertions, programmers working on this code may
not understand regression errors down the line.

Assertions have been added to catch bad errors that a
programmer editing the code may produce down the line.
e.g. uninitialized objects

Let's add assertions to catch bad programming mistakes
that might occur in the future.

LimZiJia committed:
Fix bad switch case statement in parser

Parse has a catch all default statement.

This is bad because default should be the default option
and not for the last option(s).

TODO, DEADLINE, EVENT cases have been consolidated to
call the addTask() method and default option is as intended.

Using default as the last option is an error prone way to
write code.

Please do not use default to catch remaining cases in the future.

LimZiJia committed:
Update build.gradle

gradlew ShadowJar is not working.

To build jar i need to update build.gradle.

Changed the main class to duke.ui.Launcher.

Because the wrong main class was causing ShadowJar to not
build properly.

Also fixed find() printing problem.

LimZiJia committed:
Improve code quality of parser

LimZiJia committed:
Improve code quality for tasks

LimZiJia committed:
Remove magic numbers from Main

LimZiJia committed:
Tidy code and fix bug

Overall comment for the last few commits.
Code quality is lacking.

The code needs to be maintainable and easily understandable.

Changed names to be easier to understand, improved
switch cases, improved inline comments and removed
magic numbers.

This will help people to understand the code better and
reduce the chance of bugs and misunderstandings.

Because I was combing through for code quality, I realised
a big bug for the equality of two TaskLists.

LimZiJia committed:
Merge pull request #1 from LimZiJia/branch-A-Assertions

Implement Java assertions

LimZiJia committed:
Merge branch 'master' into branch-A-CodeQuality

LimZiJia committed:
Merge pull request #2 from LimZiJia/branch-A-CodeQuality

Improve code quality

LimZiJia committed:
Add mass ops

mark, unmark and delete now accepts a list of indexes separated
by spaces.

LimZiJia committed:
Merge pull request #3 from LimZiJia/branch-BCD-Extension

Add mass ops

LimZiJia committed:
Add CI and fix tests

LimZiJia committed:
Make public static into private static with modifiers

LimZiJia committed:
Improve JavaFX

LimZiJia committed:
Create UI.png

LimZiJia committed:
Update README

LimZiJia committed:
Rename UI.png to Ui.png

LimZiJia committed:
Fix image URL
